pkg/apis/serving/k8s_validation_test.go

```diff
@@ -1632,9 +1738,18 @@ func TestContainerValidation(t *testing.T) {
 				ctx = config.ToContext(ctx, cfg)
 			}
 
-			got := ValidateContainer(ctx, test.c, test.volumes)
-			if diff := cmp.Diff(test.want.Error(), got.Error()); diff != "" {
-				t.Errorf("ValidateContainer (-want, +got): \n%s", diff)
+			if !test.skipContainerTest {
+				got := ValidateContainer(ctx, test.c, test.volumes)
+				if diff := cmp.Diff(test.want.Error(), got.Error()); diff != "" {
+					t.Errorf("ValidateContainer (-want, +got): \n%s", diff)
+				}
+			}
+
+			if test.testAsInitContainer {
+				got := validateInitContainer(ctx, test.c, test.volumes)
+				if diff := cmp.Diff(test.wantForInitContainer.Error(), got.Error()); diff != "" {
+					t.Errorf("ValidateInitContainer (-want, +got): \n%s", diff)
+				}
```

Comment:
I think for the sake of clarity and readability we pull the init container validation tests into a separate `Test` function  with it's own cases

Reply:
I thought about it but this seems the most compact representation to avoid repeating stuff as most scenarios apply both to user containers and init containers. Otherwise I will have to copy all common cases to a func that returns them and then add only the specific ones for user containers, as the init containers ones are a subset. Anyway I can try it.